src/agent/onefuzz-supervisor/src/work.rs

```diff
@@ -28,6 +30,43 @@ impl WorkSet {
     pub fn task_ids(&self) -> Vec<TaskId> {
         self.work_units.iter().map(|w| w.task_id).collect()
     }
+
+    pub fn context_path() -> Result<PathBuf> {
+        Ok(onefuzz::fs::onefuzz_root()?.join("workset_context.json"))
+    }
+
+    pub async fn load_context() -> Result<Option<Self>> {
```

Comment:
In the future, we should try to decouple these kinds of ctors from the actual type. This impl is convenient, but bakes application-wide knowledge into a `WorkSet` ctor that might be better named `load_from_onefuzz_context()` or `load_from_fs_context()`.

Reply:
Concretely, for now, how about we just rename this `load_from_context()`, since it loads a `WorkSet` from the filesystem context?

Reply:
(Ah, I see, you were mostly just copying what we do in the reboot case. There "context == workset" is an implementation detail, which is why I'd choose different language here).